codegen-sbt/src/sbt-test/codegen/test-compile-newtype/build.sbt

```diff
@@ -15,14 +15,32 @@ lazy val root = project
     libraryDependencies ++= Seq(
       "com.github.ghostdogpr" %% "caliban" % Version.pluginVersion
     ),
-    Compile / caliban / calibanSettings ++= Seq(
-      calibanSetting(file("src/main/graphql/schema.graphql"))(
-        _.genType(Codegen.GenType.Schema)
-          .clientName("GeneratedAPI")
-          .packageName("graphql")
-          .effect("MyZQuery")
-          .scalarMapping("ID" -> "String")
-      )
-    )
+    Compile / caliban / calibanSettings ++= {
+      CrossVersion.partialVersion(scalaVersion.value) match {
+        case Some((3, _)) =>
+          Seq(
+            calibanSetting(file("src/main/graphql/schema.graphql"))(
+              _.genType(Codegen.GenType.Schema)
+                .clientName("GeneratedAPI")
+                .packageName("graphql")
+                .effect("MyZQuery")
+                .scalarMapping("ID" -> "String")
+                .addDerives(true)
+                .zioEnv("graphql.Env")
```

Comment:
New CLI argument here.. passing the Env type.

Reply:
I wonder if we should name it `envForDerives` or something like that, otherwise people may be confused that it's used in the effects returned or elsewhere.